src/core/lender/test/mock/assessor.sol

```diff
@@ -1,14 +1,57 @@
 pragma solidity >=0.4.24;
 
-contract AssessorMock {
+import "ds-math/math.sol";
 
-    uint public assetReturn; function setAssetReturn(uint assetAmount_) public {assetReturn=assetAmount_;}
-    
-    uint public wad;
-    address public tranche;
+contract AssessorMock is DSMath {
 
-    function getAssetValueFor(address tranche_) public returns (uint) {
-        tranche = tranche_;
-        return assetReturn;
+    uint256 constant ONE = 10 ** 27;
+
+    address public senior;
+    address public junior;
+
+    mapping (bytes32 => uint) calls;
+    mapping (bytes32 => uint) returnValues;
+
+
+    function setReturn(bytes32 name, uint returnValue) public {
+        returnValues[name] = returnValue;
+    }
+
+    function call(bytes32 name) internal returns (uint) {
+        calls[name]++;
+        return returnValues[name];
+    }
+
+    function calcTokenPrice () public returns (uint) {
+        return call("tokenPrice");
+    }
+
+    function calcAssetValue(address tranche) public returns (uint) {
+        uint trancheReserve;
+        if (tranche == junior) {
+            trancheReserve = returnValues["jBalance"];
+        }
+        trancheReserve = returnValues["sBalance"];
+        uint poolValue = returnValues["pileDebt"];
+        if (tranche == junior) {
+            return calcJuniorAssetValue(poolValue, trancheReserve, seniorDebt());
+        }
+        return calcSeniorAssetValue(poolValue, trancheReserve, seniorDebt(), juniorReserve());
+    }
+
+    // Tranche.assets (Junior) = (Pool.value + Tranche.reserve - Senior.debt) > 0 && (Pool.value - Tranche.reserve - Senior.debt) || 0
+    function calcJuniorAssetValue(uint poolValue, uint trancheReserve, uint seniorDebt) internal returns (uint) {
+        int assetValue = int(poolValue + trancheReserve - seniorDebt);
```

Comment:
I think if this is just supposed to be a mock, it shouldn't need to implement this logic?

Reply:
maybe we should consider renaming `trancheReserve` to `juniorReserve`, wdyt?

Reply:
@manuelpolzhofer we also have a todo to find a different name for `poolValue` as we don't call it a pool anywhere else.

Reply:
> I think if this is just supposed to be a mock, it shouldn't need to implement this logic?

yea you're right, I changed it in the next PR

Reply:
> I think if this is just supposed to be a mock, it shouldn't need to implement this logic?

I think all the values are anyway passed down to the function that calls these functions, so rather than mocking the whole thing we can just use those values. wdyt? I prefixed now the method names so that it is clear they are internal.